src/Remote/Importer.php

```diff
@@ -0,0 +1,162 @@
+
+        throw new InvalidImportFormat(sprintf('The import scheme "%s" is not supported.', $scheme));
+    }
+
+    public function fetchPackages(): void
+    {
+        if (!$this->imports) {
+            $this->composer->remove();
+
+            return;
+        }
+
+        $input = GlobalHelper::getApplication()->getInput();
+
+        // Need to look for the raw options as the input is not yet parsed
+        $forceUpdate = null === $input->getParameterOption('--update-remotes', true);
+        $displayProgress = null !== $input->getParameterOption('--no-interaction', true);
+
+        $autoloadPath = PathHelper::getRoot() . Composer::VENDOR_DIR . 'autoload.php';
+
+        if (!file_exists($autoloadPath)) {
+            $forceUpdate = true;
+        }
+
+        $this->composer->update($forceUpdate, $displayProgress);
+
+        require_once $autoloadPath;
+
+        foreach ($this->imports as $package => $data) {
+            foreach ($data as $version => $files) {
+                foreach ($files as $file) {
+                    import(PathHelper::getRoot() . Composer::VENDOR_DIR . $package . '/' . ($file ?? ''));
+                }
+            }
+        }
+    }
+
+    /**
+     * @param ?array{
+     *     url?: string,
+     *     type?: "git" | "svn",
+     *     reference?: string,
+     * } $source
+     */
+    private function importPackageWithComposer(string $package, string $version, ?string $repositoryUrl = null, ?array $source = null, ?string $file = null): void
+    {
+        $this->logger->notice('Importing remote package with Composer.', [
+            'package' => $package,
+            'version' => $version,
+        ]);
+
+        $json = $this->composer->getConfiguration();
+
+        $json['require'][$package] = $version;
+
+        if ($repositoryUrl) {
+            $json['repositories'][] = [
+                'type' => 'vcs',
+                'url' => $repositoryUrl,
+            ];
+        }
+
+        if ($source) {
+            if (!isset($source['url'], $source['type'], $source['reference'])) {
+                throw new ImportError('The "source" argument must contain "url", "type" and "reference" keys.');
+            }
+
+            $json['repositories'][] = [
+                'type' => 'package',
+                'package' => [
+                    'name' => $package,
+                    'version' => $version,
+                    'source' => $source,
+                ],
+            ];
+        }
+
+        $this->composer->setConfiguration($json);
+
+        $this->imports[$package][$version][] = $file;
+    }
+
+    private function allowsRemote(): bool
+    {
+        if ($_SERVER['CASTOR_NO_REMOTE'] ?? false) {
+            return false;
+        }
+
+        $input = GlobalHelper::getApplication()->getInput();
+
+        // Need to look for the raw options as the input is not yet parsed
+        $noRemote = $input->getParameterOption('--no-remote', true);
+
+        return true === $noRemote;
```

Comment:
Not exactly the same. In case you pass `--no-remote --foobar` (so anything after `--no-remote`), the returned value for the **raw** `--option` will be `--foobar`, so a true-ish value. Here is an example when dumping the raw option:

```bash
$ castor
^ true
$ castor --no-remote
^ null
$ castor --no-remote -v
^ "-v"
```